pkg/render/guardian.go

```diff
@@ -368,6 +430,58 @@ func (c *GuardianComponent) annotations() map[string]string {
 	return annotations
 }
 
+func (c *GuardianComponent) deploymentSelector() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app.kubernetes.io/name": GuardianDeploymentName,
+		},
+	}
+}
+
+const (
+	GoldmaneDeploymentName = "goldmane"
+	GoldmaneNamespace      = common.CalicoNamespace
+)
+
+func (c *GuardianComponent) networkPolicy() *netv1.NetworkPolicy {
+	return &netv1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "networking.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "allow-whisker", Namespace: GuardianNamespace},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: *c.deploymentSelector(),
+			PolicyTypes: []netv1.PolicyType{netv1.PolicyTypeIngress},
+			Ingress: []netv1.NetworkPolicyIngressRule{
+				{
+					From: []netv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"app.kubernetes.io/name": GoldmaneDeploymentName,
+								},
+							},
+							NamespaceSelector: &metav1.LabelSelector{
```

Comment:
Same here for namespace selector - we can get rid of it when they are in the same namespace.